### PR TITLE
Generalize retrieve databundle

### LINKED ISSUE #259: Generalize retrieve_databundle to the world
## Generalize retrieve_databundle to the world

Currently, retrieve_databundle works only for Africa in a relatively hard-fix way.
There is the need to give it a more customizable structure to enable users do successfully download the required inputs depending on the selected countries

### PR DESCRIPTION
**Pull request by davide-f**
# Closes # #259 

## Changes proposed in this Pull Request

This pull request aims at providing a general retrieve_databundle structure that enables a flexible addition of diversified bundles for different regions.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes.

### CODE REVIEW

**Comment by davide-f** [comment]
The pull request is already in good shape, yet, I would like to improve how it works.
Currently, for every category [resources, data, cutouts, common], it downloads a single databundle, yet in general, since we may want a databundle per continent (if not more):
- [x] the procedure shall download more databundles per category.

Moreover, debugging is needed including the appropriate urls and we may need to re-organize the databundles for consistency. In particular:
- [x] need to merge common world-level data [e.g. GDP raster] to a single zip file
- [x] load to zenodo only zip files

**Comment by davide-f** [comment]
@pz-max I think this PR is at an advanced stage and the CI works, feel free to add comments on them.
For testing zenodo, I created a sandbox repo in https://sandbox.zenodo.org/record/1016042#.Yg0vL9_MJPY
This test sandbox shall be removed once everything is fixed and we shall move all zip files into the official one

File: scripts/build_renewable_profiles.py [hunk 1]
```diff
@@ -252,13 +252,18 @@
 
     # filter plants for hydro
     if snakemake.wildcards.technology.startswith("hydro"):
+        country_shapes = gpd.read_file(paths.country_shapes)
         hydrobasins = gpd.read_file(resource["hydrobasins"])
-        merged_hydrobasin_shape = unary_union(hydrobasins.geometry)
+        hydrobasins = hydrobasins[[
```
Comment (pz-max) [inline scripts/build_renewable_profiles.py:257]: Can you linger on the reason for this change? What was the previous issue?

File: scripts/retrieve_databundle_light.py [hunk 2]
```diff
@@ -32,17 +32,189 @@
...
+        logger.info(f"Download resource '{resource}' from cloud '{url}'.")
+
+        return True
+    else:
+        logger.error(f"Host {host} not implemented")
+        return False
+
+
+"""
+    get_best_bundles(country_list, category, config_bundles, tutorial)
```
Comment (pz-max) [inline scripts/retrieve_databundle_light.py]: I think this docstring is misplaced?

File: scripts/build_renewable_profiles.py (hunk 1 above)
Comment (davide-f) [reply scripts/build_renewable_profiles.py:257]: The unary_union for the world hydrobasins was taking forever to compute.
Therefore, I preferred to decompose the problem into:
1. remove from hydrobasins dataframe the shapes that do not intersect to the countries of interest
2. perform the computations for only the buses that lay in a hydrobasin shape

File: scripts/retrieve_databundle_light.py (hunk 2 above)
Comment (davide-f) [reply scripts/retrieve_databundle_light.py]: Thanks! modified

**Comment by davide-f** [comment]
I've just noted that the resource files from gegis are just the raw files; they don't have the fix for the empty countries, thus the known issue arise.
- [x] Need to update the raw files in drive and zenodo

Moreover:
- [ ] Once settled, and we agree on merging, we shall move the zenodo files from sandbox [the test repo I've used] to the official one

**Comment by davide-f** [comment]
I filled the databundles the "empty" Africa countries demand data for all years [2030, 2040, 2050 and 2100] with era5 data 2013, and updated all databundles [zenodo&googledrive] accordingly.

This branch may be merged as-is and it should work; anyway, we may need to use the links to the official PyPSA-Earth zenodo.
The easy way if all zip files are moved from one zenodo to the other one

**Comment by davide-f** [comment]
Tests performed:
- [x] Turkey test (with manual change of input file Africa.nc to Asia.nc as input file to add_electricity)
- [x] Africa test

Once both tests are complete (expected soon), I think this PR could be ready to merge, however, we shall update the official Zenodo; @pz-max I'm not sure if you can add permissions for pushing file to Zenodo, otherwise, you may just download and upload all zip files available (except resources_africa that is a duplication) at [sandbox Zenodo](https://sandbox.zenodo.org/record/1016878).
As a little comment, resources_world contains demand data for the whole world, but the natura.tiff is calibrated only for africa.
I may update that file, but also the current version is file as natura.tiff can be created by enabling build_cutout.

**Comment by pz-max** [comment]
Awesome! Regarding Zenodo, I made a fatal error. I uploaded everything from my Github linked account. Unfortunately, I cannot give anyone edit access or something like that. Suggestion. I created now a new gmail account called pypsameetsearth@gmail.com. We register a Zenodo account over it so that everyone can upload data?

Login details can be found in the trusted coordinator discord channel

**Comment by davide-f** [comment]
@pz-max I totally agree on creating the Zenodo account for the organization. Add some comments in discord

**Comment by pz-max** [comment]
@davide-f  can this be merged?

**Comment by davide-f** [comment]
I think this PR can be merged; though there can be improvements but postponed into future PRs and commits.
In particular, I foresee:
- [ ] automatically download the eez zones
- [ ] automatically download the WPDA data
- [ ] add license package to the documentation

**Pull request merged**